doc improvement: doc cleanup mesh dfu (#11209)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/dfu/target/README.rst

* Config Server allows configurator devices to configure the node remotely.
* Health Server provides ``attention`` callbacks that are used during provisioning to call your attention to the device.
  These callbacks trigger blinking of the LEDs.

> * Use of Binary Large Object (BLOB) Transfer models is the underlying transport method for the DFU.

Reviewer: ```suggestion
* Binary Large Object (BLOB) Transfer models are the underlying transport mechanism for the mesh DFU feature.
```
